packages/lesswrong/server/languageModels/modGPT.ts

```diff
@@ -110,39 +115,35 @@ async function checkModGPT(comment: DbComment, post: DbPost): Promise<void> {
     return
   }
   
-  if (!comment.contents?.originalContents?.data) {
+  if (!comment.contents?.originalContents?.data || !post.contents?.originalContents?.data) {
     if (!isAnyTest) {
       //eslint-disable-next-line no-console
       console.log("Skipping ModGPT (no contents on this comment!)")
     }
     return
   }
 
-  const commentText = sanitizeHtml(comment.contents?.html ?? "", {
-    allowedTags: difference(sanitizeAllowedTags, ['img', 'iframe', 'audio']),
-    nonTextTags: [ 'style', 'script', 'textarea', 'option', 'img' ]
-  })
-  const postText = sanitizeHtml(post.contents?.html ?? "", {
-    allowedTags: difference(sanitizeAllowedTags, ['img', 'iframe', 'audio']),
-    nonTextTags: [ 'style', 'script', 'textarea', 'option', 'img' ]
-  })
+  const commentHtml = await dataToHTML(comment.contents.originalContents.data, comment.contents.originalContents.type, {sanitize: false})
+  const postHtml = await dataToHTML(post.contents.originalContents.data, post.contents.originalContents.type, {sanitize: false})
+  const commentText = sanitizeHtml(commentHtml ?? "", sanitizeHtmlOptions)
+  const postText = sanitizeHtml(postHtml ?? "", sanitizeHtmlOptions)
```

Comment:
The main issue was that `contents.html` wasn't always available, in particular when a comment was edited.